src/main/java/core/basesyntax/converter/StringToFruitDaoConverter.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.dao.FruitDao;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class StringToFruitDaoConverter implements Function<String, FruitDao> {
+
+    @Override
+    public FruitDao apply(String s) {
+        StringToFruitValidator validator = new StringToFruitValidator();
+        String[] arr = s.split(",");
+        if (!validator.validateInputData(arr)) {
+            throw new IllegalArgumentException("Error while parsing string [" + s + "]");
```

Comment:
Ensure that exceptions are handled properly and provide meaningful messages to help with debugging. Consider logging the exception or providing more context in the error message.